src/api/totalIssuance.ts

```diff
@@ -1,39 +1,23 @@
 import { ApiPromise } from "@polkadot/api"
-import { useQueries, useQuery } from "@tanstack/react-query"
+import { useQuery } from "@tanstack/react-query"
 import { QUERY_KEYS } from "utils/queryKeys"
 import BigNumber from "bignumber.js"
-import { u32 } from "@polkadot/types"
-import { Maybe, undefinedNoop } from "utils/helpers"
-import { AnyNumber } from "@polkadot/types-codec/types"
 import { useRpcProvider } from "providers/rpcProvider"
 
-export const useTotalIssuance = (token: Maybe<u32 | string | AnyNumber>) => {
-  const { api } = useRpcProvider()
+export const useTotalIssuances = () => {
+  const { isLoaded, api } = useRpcProvider()
 
-  return useQuery(
-    QUERY_KEYS.totalIssuance(token),
-    token ? getTotalIssuance(api, token) : undefinedNoop,
-    { enabled: !!token },
-  )
+  return useQuery(QUERY_KEYS.totalIssuances, getTotalIssuances(api), {
+    enabled: isLoaded,
+  })
 }
 
-export const useTotalIssuances = (tokens: Maybe<u32 | string>[]) => {
-  const { api } = useRpcProvider()
+const getTotalIssuances = (api: ApiPromise) => async () => {
+  const res = await api.query.tokens.totalIssuance.entries()
 
-  const tokenIds = tokens.filter((token): token is u32 => !!token)
+  return res.reduce<Map<string, BigNumber>>((acc, [key, rawData]) => {
```

Comment:
```suggestion
  return new Map(
    res.map(([key, rawData]) => [
      key.args[0].toString(),
      rawData.toBigNumber(),
    ]),
  )
```

I think this is cleaner